File: pset6/readability/readability.py
from cs50 import get_string

# Prompt user for input
text = get_string("Text: ")

# Get number of letters, words, and sentences in text
letters, words, sentences = 0, 1, 0
for char in text:
    if char.isalpha():
        letters += 1
    elif char == " ":
        words += 1
    elif char in ["!", "?", "."]:
        sentences += 1

# Get L ans S per Coleman-Liau index
L = 100 * letters / words
S = 100 * sentences / words

# Calculate the index
index = round(0.0588 * L - 0.296 * S - 15.8)

# Print the text's grade
if index > 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)
{
    // Prompt user for input
    string text = get_string("Text: ");
    
    // Get number of letters, words, and sentences in text
    int letters = 0, words = 1, sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else if (text[i] == ' ')
        {
            words++;
        }
        else if (text[i] == '!' || text[i] == '.' || text[i] == '?')
        {
            sentences++;
        }
    }
    
    // Get L ans S per Coleman-Liau index
    float L = 100.0 * letters / words;
    float S = 100.0 * sentences / words;
    
    // Calculate the index
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    
    // Print the text's grade
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}